### PR TITLE
Remove mandatory generated files on windows too

### PR DESCRIPTION
**Pull request by bernd-edlinger**
There is still one more file lying around: test\uitest.lib
I don't know where that came from.

### CODE REVIEW

File: Configurations/windows-makefile.tmpl [hunk 1]
```diff
@@ -360,10 +360,12 @@ clean: libclean
 	{- join("\n\t", map { "-del /Q /F $_" } @PROGRAMS) -}
 	-del /Q /F $(ENGINES)
 	-del /Q /F $(SCRIPTS)
+	-del /Q /F $(GENERATED_MANDATORY)
```
Comment (levitte) [inline Configurations/windows-makefile.tmpl]: Please add the same in `Configurations/unix-Makefile.tmpl` as well.  If you dare, you can also add this line in `Configurations/descrip.mms.tmpl`:

``` make
        {- join("\n\t", map { "- DELETE $_;*" } @{$unified_info{depends}->{""}}) || "@ !" -}
```

**Comment by dot-asm** [comment]
One can wonder if clean: is the right spot for it. Or rather question I wonder is if distclean: would be more appropriate? Rationale is that GENERATED_MANDATORY effectively defines the target and in this quality is tied to configdata.pm, and since configdata.pm is removed in distclean... In a sense configdata.pm is most "mandatory" of them all...

File: Configurations/windows-makefile.tmpl (hunk 1 above)
Comment (dot-asm) [reply Configurations/windows-makefile.tmpl]: It was done for for unix-Makefile already under cygwin disguise.

**Comment by levitte** [comment]
> Or rather question I wonder is if distclean: would be more appropriate?

The "standard" idea with `distclean` is to remove everything that `Configure` created.  The generated files are created from `Makefile` rules, so from that point of view, the answer is "no".

**Comment by bernd-edlinger** [comment]
Added same logic to VMS.

Note that occasionally the Configure script uses context info from the host machine
to determine what is installed on the build machine:
```
$ ./Configure  VC-WIN32
Configuring OpenSSL version 1.1.1-pre5-dev (0x10101005L) for VC-WIN32
Using os-specific seed configuration
NASM not found - please read INSTALL and NOTES.WIN for further details
$ ./Configure  VC-WIN32 no-asm
Configuring OpenSSL version 1.1.1-pre5-dev (0x10101005L) for VC-WIN32
Using os-specific seed configuration

******************************************************************************
This perl implementation doesn't produce Windows like paths (with backward
slash directory separators).  Please use an implementation that matches your
building platform.

This Perl version: 5.18.2 for x86_64-linux-gnu-thread-multi
******************************************************************************
```

File: Configurations/windows-makefile.tmpl (hunk 1 above)
Comment (bernd-edlinger) [reply Configurations/windows-makefile.tmpl]: done.

**Comment by FdaSilvaYY** [comment]
There is currently a bug in the `nmake clean` : it delete all the `.exp|.export|.expected` files from the krb5 folder who is a git sub-module .

**Comment by bernd-edlinger** [comment]
Well, this line deletes recursively all .exp but it does not look like it should match .export/.expected:
`-del /Q /S /F *.d *.obj *.pdb *.exp *.ilk *.manifest`

I'd say that is for another PR.

**Comment by bernd-edlinger** [comment]
Merged.  Thanks!

**Pull request closed without merging**